### PR TITLE
feat(chat): confirmation message if imports chat prompt is exist (Issue #609, Issue #1195)

### PR DESCRIPTION
**Pull request by denys-kolomiitsev**
**Description:**

Confirmation message if imports chat prompt is exist.
Fix import of the folder with the dot in the end of name.

Issues:

- Issue #609 
- Issue #1195 

**UI changes**

image

**Checklist:**

- [x] the pull request name complies with [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/)
- [x] the pull request name starts with `fix(<scope>):`, `feat(<scope>):`, `feature(<scope>):`, `chore(<scope>):`, `hotfix(<scope>):` or `e2e(<scope>):`. If contains breaking changes then the pull request name must start with `fix(<scope>)!:`, `feat(<scope>)!:`, `feature(<scope>)!:`, `chore(<scope>)!:`, `hotfix(<scope>)!:` or `e2e(<scope>)!:` where `<scope>` is name of affected project: `chat`, `chat-e2e`, `overlay`, `shared`, `sandbox-overlay`, etc.
- [x] the pull request name ends with `(Issue #<TICKET_ID>)` (comma-separated list of issues)
- [x] I confirm that do not share any confidential information like API keys or any other secrets and private URLs

### CODE REVIEW

**Comment by ai-dial-actions** [comment]
Hey there and thank you for opening this pull request! 👋🏼

We require pull request titles to follow the [Conventional Commits specification](https://www.conventionalcommits.org/en/v1.0.0/) and it looks like your proposed title needs to be adjusted.

Details:

```
No release type found in pull request title "feat(chat):609 confirmation message if imports chat prompt is exist". Add a prefix to indicate what kind of release this pull request corresponds to. For reference, see https://www.conventionalcommits.org/

Available types:
 - fix
 - hotfix
 - feat
 - feature
 - chore
```

File: apps/chat/src/utils/app/import-export.ts [hunk 1]
```diff
@@ -446,3 +449,29 @@ export const updateAttachment = ({
...
+    case FeatureType.Prompt:
+      return ImportExportActions.cancelPromptsImportDuplicated;
+    //TODO implement case FeatureType.File
+
+    default:
+      return ImportExportActions.cancelConversationsImportDuplicated;
+  }
+};
+
+export const getFolderType = (featureType: FeatureType) => {
```
Comment (IlyaBondar) [inline apps/chat/src/utils/app/import-export.ts]: move it in `mappers.ts` as `getFolderTypeByFeatureType`

File: apps/chat/src/utils/app/common.ts [hunk 2]
```diff
@@ -41,6 +41,17 @@ export const isEntityNameOnSameLevelUnique = (
   return !sameLevelEntities.some((e) => nameToBeUnique === e.name);
 };
 
+export const isImportEntityNameOnSameLevelUnique = (
```
Comment (IlyaBondar) [inline apps/chat/src/utils/app/common.ts]: we already have `isEntityNameOnSameLevelUnique`

File: apps/chat/src/utils/app/clean.ts [hunk 3]
```diff
@@ -78,9 +79,12 @@ export const cleanConversation = (
       conversation.id ||
       constructPath(
         conversation.folderId || getConversationRootId(),
-        conversation.name || DEFAULT_CONVERSATION_NAME,
+        (conversation.name && prepareEntityName(conversation.name)) ||
```
Comment (IlyaBondar) [inline apps/chat/src/utils/app/clean.ts]: we shouldn't use `prepareEntityName` into this method - unexpected update name can happen when we saved other. Use it directly in your high-level method

File: apps/chat/src/types/import-export.ts [hunk 4]
```diff
@@ -83,3 +83,18 @@ export enum ImportRoot {
   Imports = 'imports',
   Files = 'files',
 }
+
+export enum LoadingStatus {
```
Comment (IlyaBondar) [inline apps/chat/src/types/import-export.ts]: use `UploadStatus` instead

File: apps/chat/src/utils/app/common.ts (hunk 2 above)
Comment (denys-kolomiitsev) [reply apps/chat/src/utils/app/common.ts]: This is another method. `isEntityNameOnSameLevelUnique` not working in this case.

File: apps/chat/src/utils/app/import-export.ts (hunk 1 above)
Comment (denys-kolomiitsev) [reply apps/chat/src/utils/app/import-export.ts]: Thank you for the suggestion!
Method moved to the mappers.ts

File: apps/chat/src/types/import-export.ts (hunk 4 above)
Comment (denys-kolomiitsev) [reply apps/chat/src/types/import-export.ts]: Changed to use `UploadStatus`

File: apps/chat/src/utils/app/clean.ts (hunk 3 above)
Comment (denys-kolomiitsev) [reply apps/chat/src/utils/app/clean.ts]: Moved it one level up. Thanks for the advise!

**Comment by denys-kolomiitsev** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-1136.nightly-test.deltixhub.io
>E2E tests status: failed

**Comment by denys-kolomiitsev** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-1136.nightly-test.deltixhub.io
>E2E tests status: success

File: apps/chat/src/components/Common/ReplaceConfirmationModal/Components.tsx [hunk 5]
```diff
@@ -0,0 +1,260 @@
...
+  },
+  {
+    id: ReplaceOptions.Replace,
+    displayName: ReplaceOptions.Replace,
+  },
+  {
+    id: ReplaceOptions.Ignore,
+    displayName: ReplaceOptions.Ignore,
+  },
+];
```
Comment (IlyaBondar) [inline apps/chat/src/components/Common/ReplaceConfirmationModal/Components.tsx]:
```suggestion
const replaceSelectorOptions: SelectOption[] = [
    ReplaceOptions.Postfix, 
    ReplaceOptions.Replace, 
    ReplaceOptions.Ignore
].map(option => ({
    id: option,
    displayName: option
}));
```

File: apps/chat/src/components/Common/ReplaceConfirmationModal/Components.tsx [hunk 6]
```diff
@@ -0,0 +1,260 @@
...
+import { useAppSelector } from '@/src/store/hooks';
+import { ModelsSelectors } from '@/src/store/models/models.reducers';
+
+import { ModelIcon } from '../../Chatbar/ModelIcon';
+import { Select, SelectOption } from '../Select';
+import Tooltip from '../Tooltip';
+
+interface ReplaceSelectorProps {
+  selectedOption: string;
+  onOptionChangeHandler: (option: string) => void;
```
Comment (IlyaBondar) [inline apps/chat/src/components/Common/ReplaceConfirmationModal/Components.tsx]:
```suggestion
  selectedOption: ReplaceOptions;
  onOptionChangeHandler: (option: ReplaceOptions) => void;
```

File: apps/chat/src/components/Common/ReplaceConfirmationModal/ConversationsList.tsx [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+          </div>
+        );
+      })}
+      {conversationsToReplace.map((conversation, index) => {
+        if (!isRootId(conversation.folderId)) {
+          return null;
+        }
+
+        return (
+          <div key={index + conversation.id}>
```
Comment (Alexander-Kezik) [inline apps/chat/src/components/Common/ReplaceConfirmationModal/ConversationsList.tsx]: Wouldn't the key be unique without an index?

File: apps/chat/src/components/Common/Select.tsx [hunk 8]
```diff
@@ -0,0 +1,86 @@
...
+        }
+      >
+        {options.map((option) => (
+          <MenuItem
+            key={option.id}
+            className={classNames(
+              'max-w-[350px] hover:bg-accent-primary-alpha',
+              optionClassName,
+            )}
+            item={t(option.displayName)}
```
Comment (Alexander-Kezik) [inline apps/chat/src/components/Common/Select.tsx]: I think translation should be moved to the `options` to apply translation properly?

File: apps/chat/src/components/Common/ReplaceConfirmationModal/ConversationsList.tsx (hunk 7 above)
Comment (denys-kolomiitsev) [reply apps/chat/src/components/Common/ReplaceConfirmationModal/ConversationsList.tsx]: Thank you for the suggestion. `Index` removed.

File: apps/chat/src/components/Common/ReplaceConfirmationModal/Components.tsx (hunk 5 above)
Comment (denys-kolomiitsev) [reply apps/chat/src/components/Common/ReplaceConfirmationModal/Components.tsx]: Thank you for the suggestion. 
Changed as advised.

File: apps/chat/src/components/Common/ReplaceConfirmationModal/Components.tsx (hunk 6 above)
Comment (denys-kolomiitsev) [reply apps/chat/src/components/Common/ReplaceConfirmationModal/Components.tsx]: Thank you for the suggestion. 
`selectedOption` changed.
`onOptionChangeHandler` was not changed as it depends on undelying component props and can recive only this type `(option: string) => void;`

File: apps/chat/src/components/Common/Select.tsx (hunk 8 above)
Comment (denys-kolomiitsev) [reply apps/chat/src/components/Common/Select.tsx]: Thank you for the suggestion.
Translation moved to the `options`.

**Comment by denys-kolomiitsev** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-1136.nightly-test.deltixhub.io
>E2E tests status: failed

**Review by Alexander-Kezik — DISMISSED** [review]

**Comment by denys-kolomiitsev** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-1136.nightly-test.deltixhub.io
>E2E tests status: failed

**Comment by denys-kolomiitsev** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-1136.nightly-test.deltixhub.io
>E2E tests status: failed

**Comment by denys-kolomiitsev** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-1136.nightly-test.deltixhub.io
>E2E tests status: failed

**Comment by denys-kolomiitsev** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-1136.nightly-test.deltixhub.io
>E2E tests status: failed

**Comment by denys-kolomiitsev** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-1136.nightly-test.deltixhub.io
>E2E tests status: failed

**Comment by denys-kolomiitsev** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-1136.nightly-test.deltixhub.io
>E2E tests status: success

**Review by Alexander-Kezik — APPROVED** [review]

**Pull request merged**